### ISSUE TITLE
Fails with latest `cargo-generate`

### ISSUE THREAD

**Issue by orhun** [rustwasm/wasm-pack#1245]
## 🐛 Bug description

I'm using cargo-generate from [Arch repositories](https://archlinux.org/packages/community/x86_64/cargo-generate/):

```sh
$ halp --no-help cargo-generate
(°ロ°)  checking 'cargo-generate -V'
\(^ヮ^)/ success '-V' argument found!
cargo generate 0.18.1
```

I just built `wasm-pack` `0.11.0` and it fails like the following:

```sh
$ wasm-pack new test
Error: expected value at line 1 column 1
Caused by: expected value at line 1 column 1
```

However, if I uninstall `cargo-generate` and let `wasm-pack` handle that:

```sh
$ pacman -R cargo-generate
$ wasm-pack new test
[INFO]: ⬇️  Installing cargo-generate...
🐑  Generating a new rustwasm project with name 'test'...
```

It works as expected.

I tracked down this issue with using `strace` and found out `wasm-pack` actually installs an older version of `cargo-generate`:

```sh
$ /home/orhun/.cache/.wasm-pack/cargo-generate-9349d7b81730c28d/cargo-generate -V
cargo generate 0.17.3
```

Thus I concluded this is something that has changed between the latest version of `cargo-generate` and the version that is used by `wasm-pack`. 

#### 🤔 Expected Behavior
Successful run.

#### 👟 Steps to reproduce
1. Install `cargo-generate` (`pacman -S cargo-generate`)
2. [Build](https://wiki.archlinux.org/title/Makepkg) `0.11.0` and install `wasm-pack` using this [`PKGBUILD`](https://github.com/archlinux/svntogit-community/blob/packages/wasm-pack/trunk/PKGBUILD).

#### 🌍 Your environment
wasm-pack version: `0.11.0`
rustc version: `rustc 1.68.0 (2c8cc3432 2023-03-06) (Arch Linux rust 1:1.68.0-1)`

**Comment by drager** [comment]
You are correct. We are downloading `0.17.3` in wasm-pack: https://github.com/rustwasm/wasm-pack/blob/d5cb84e3e802293154a1b85db310cbb458b5a941/src/install/mod.rs#L194-L200

I suppose we can upgrade that and debug why it's failing then.

**Comment by sassman** [comment]
I guess under the hood it's just running something like:

```sh
cargo generate rustwasm/wasm-pack-template
```

I tried this successfully with the latest version of cargo-generate `v0.18.2` on a M1.
<img width="1027" alt="image" src="https://user-images.githubusercontent.com/329682/231711380-ebfe2400-5688-4fe3-9c5f-4748972ef16d.png">

**Comment by orhun** [comment]
This is magically fixed in `0.11.1`, maybe it was related to #1254
